pandas/core/arrays/string_arrow.py

```diff
@@ -0,0 +1,619 @@
+        scalars = lib.ensure_string_array(
+            scalars, na_value=cls._dtype.na_value, copy=False
+        )
+        return cls(pa.array(scalars, type=pa.string(), from_pandas=True))
+
+    @property
+    def dtype(self) -> ArrowStringDtype:
+        """
+        An instance of 'ArrowStringDtype'.
+        """
+        return self._dtype
+
+    def __array__(self, dtype=None) -> np.ndarray:
+        """Correctly construct numpy arrays when passed to `np.asarray()`."""
+        return self.to_numpy(dtype=dtype)
+
+    def __arrow_array__(self, type=None):
+        """Convert myself to a pyarrow Array or ChunkedArray."""
+        return self.data
+
+    def to_numpy(
+        self, dtype=None, copy: bool = False, na_value=lib.no_default
+    ) -> np.ndarray:
+        """
+        Convert to a NumPy ndarray.
+        """
+        # TODO: copy and na_value arguments are ignored
+        return self.data.__array__(dtype=dtype)
+
+    def __len__(self) -> int:
+        """
+        Length of this array.
+
+        Returns
+        -------
+        length : int
+        """
+        return len(self.data)
+
+    @classmethod
+    def _from_sequence_of_strings(cls, strings, dtype=None, copy=False):
+        return cls._from_sequence(strings, dtype=dtype, copy=copy)
+
+    @classmethod
+    def _from_factorized(cls, values, original):
+        return cls._from_sequence(values)
+
+    @classmethod
+    def _concat_same_type(cls, to_concat) -> ArrowStringArray:
+        """
+        Concatenate multiple ArrowStringArray.
+
+        Parameters
+        ----------
+        to_concat : sequence of ArrowStringArray
+
+        Returns
+        -------
+        ArrowStringArray
+        """
+        return cls(
+            pa.chunked_array(
+                [array for ea in to_concat for array in ea.data.iterchunks()]
+            )
+        )
+
+    def __getitem__(self, item: Any) -> Any:
+        """Select a subset of self.
+
+        Parameters
+        ----------
+        item : int, slice, or ndarray
+            * int: The position in 'self' to get.
+            * slice: A slice object, where 'start', 'stop', and 'step' are
+              integers or None
+            * ndarray: A 1-d boolean NumPy ndarray the same length as 'self'
+
+        Returns
+        -------
+        item : scalar or ExtensionArray
+
+        Notes
+        -----
+        For scalar ``item``, return a scalar value suitable for the array's
+        type. This should be an instance of ``self.dtype.type``.
+        For slice ``key``, return an instance of ``ExtensionArray``, even
+        if the slice is length 0 or 1.
+        For a boolean mask, return an instance of ``ExtensionArray``, filtered
+        to the values where ``item`` is True.
+        """
+        item = check_array_indexer(self, item)
+
+        if isinstance(item, abc.Iterable):
+            if not is_array_like(item):
+                item = np.array(item)
```

Comment:
no new test failures from removing this. maybe @xhochy can confirm it can be removed.

have removed for now, and refactored to use `is_list_like` instead of `isinstance(item, abc.Iterable)` and moved `check_array_indexer` inside the `is_list_like` condition since `check_array_indexer` is a no-op if not `is_list_like`

Reply:
> no new test failures from removing this. maybe @xhochy can confirm it can be removed.

In fletcher it was needed, because it couldn't rely yet on `check_array_indexer` being available. But so in pandas, we always have it available, so the additional manual check is not needed.

> and refactored to use is_list_like instead of isinstance(item, abc.Iterable) and moved check_array_indexer inside the is_list_like condition since check_array_indexer is a no-op if not is_list_like

Hmm, I would move it back (it's not really how it is used elsewhere in pandas, and it is explicitly a no-op for non-list-likes, so you don't need to put this behind a `if is_list_like` check). 
Also, afterwards we are sure that any list like was already converted to a numpy array, so we don't need to do an additional `is_list_like`, we can do a more precise check.

Reply:
reverted